src/App.tsx

```diff
@@ -1,61 +1,124 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { UserMas } from './interfaces/UserMas';
+import { TodosMas } from './interfaces/TodosMas';
+import { TodoList } from './components/TodoList';
+
+function getTodos(todoUserId: number): UserMas | undefined {
+  return usersFromServer.find(user => user.id === todoUserId);
+}
+
+function getNewPostId(posts: TodosMas[]) {
+  const maxId = Math.max(...posts.map(post => post.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+
+  const [isUserId, setIsUserId] = useState(0);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setIsUserId(+event.target.value);
+    setIsUserError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const [posts, setPosts] = useState<TodosMas[]>(todosFromServer);
+
+  const visibleArray = posts.map(todo => ({
+    ...todo,
+    user: getTodos(todo.userId),
+  }));
+
+  const reset = () => {
+    setTitle('');
+    setIsUserId(0);
+  };
+
+  const addPost = (newTitle: string, newUserId: number) => {
+    setPosts(currentPosts => [
+      ...currentPosts,
+      {
+        id: getNewPostId(posts),
+        title: newTitle,
+        completed: false,
+        userId: newUserId,
+      },
+    ]);
+
+    reset();
+  };
+
+  const handleSumbit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (isUserId === 0) {
+      setIsUserError(true);
+    }
+
+    if (title === '') {
```

Comment:
```suggestion
    if (!isUserId) {
      setIsUserError(true);
    }

    if (!title) {
```
You don't need to compare with an empty string and create variables as you compare them twice